py3status/modules/systemd_suspend_inhibitor.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+Turn on and off systemd suspend inhibitor.
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module; for xfce4-notifyd
+        (default 30)
+    format: display format for this module
+        (default '[\?color=state&show SUSPEND [\?if=state OFF|ON]]')
+    thresholds: specify color thresholds to use
+        (default [(True, 'bad'), (False, 'good')])
+
+Format placeholders:
+    {state} systemd suspend inhibitor state, eg True, False
+
+Color thresholds:
+    xxx: print a color based on the value of `xxx` placeholder
+```
+
+
+Examples:
+# display SUSPEND ON/OFF
+systemd_suspend_inhibitor {
+    format = '[\?color=state&show SUSPEND [\?if=state OFF|ON]]'
+    thresholds = [(True, "bad"), (False, "good")]
+}
+```
+
+@author Cyrinux https://github.com/cyrinux
+@license BSD
+
+SAMPLE OUTPUT
+[{'full_text': 'SUSPEND ON', 'color': '#00FF00'}]
+[{'full_text': 'SUSPEND OFF', 'color': '#FF0000'}]
+"""
+
+from os import close
+
+
+class Inhibit:
+    """
+    """
+
+    def __init__(self):
+        from dbus import SystemBus
+
+        self.bus = SystemBus()
+        self.fd = None
+        self.proxy = self.bus.get_object(
+            "org.freedesktop.login1", "/org/freedesktop/login1"
+        )
+
+    def get_state(self):
+        return True if self.fd else False
+
+    def toggle(self):
+        if self.fd is None:
+            self.fd = self.proxy.Inhibit(
+                "handle-lid-switch:idle:sleep",
+                "Py3Status",
+                "systemd suspend inhibitor module",
+                "block",
+                dbus_interface="org.freedesktop.login1.Manager",
+            ).take()
+        else:
+            close(self.fd)
+            self.fd = None
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 30
+    format = "[\?color=state&show SUSPEND [\?if=state OFF|ON]]"
+    thresholds = [(True, "bad"), (False, "good")]
+
+    def post_config_hook(self):
+        self.inhibit = Inhibit()
+        self.state = None
+        self.thresholds_init = self.py3.get_color_names_list(self.format)
+
+    def systemd_suspend_inhibitor(self):
+        self.state = self.inhibit.get_state()
+        data = {"state": int(self.state)}
```

Comment:
I see no reason for `self.state`. Could use `state` instead.

Reply:
Ok for bus and i rename proxy to login1 because its the object with whom I speak.